Public/Src/Engine/Scheduler/Scheduler.cs

```diff
@@ -5482,6 +5484,20 @@ SealDirectoryKind IFileContentManagerHost.GetSealDirectoryKind(DirectoryArtifact
             return GetSealDirectoryKind(directory);
         }
 
+        [SuppressMessage("Microsoft.Design", "CA1033:InterfaceMethodsShouldBeCallableByChildTypes")]

```

Comment:
Why implement this function here but not in FileConternManager.cs?

Reply:
Because `FileContentManager` doesn't know if a directory is a source sealed directory; only scheduler knows, and the functionality has been implemented as `IsSourceSealedDirectory` of `IPipExecutionEnvironment`.

---
In reply to: [330658466](https://github.com/microsoft/BuildXL/pull/977#discussion_r330658466) [](ancestors = 330658466)